cli/internal/plugin/spec/config/v2alpha1/config.go

```diff
@@ -0,0 +1,131 @@
+)
+
+const (
+	// ConfigType defines the type identifier for credential configurations
+	ConfigType = "plugin.config.ocm.software"
+)
+
+var scheme = runtime.NewScheme()
+
+func init() {
+	scheme.MustRegisterWithAlias(&Config{}, runtime.NewVersionedType(ConfigType, Version))
+}
+
+// Config represents the top-level configuration for the plugin manager.
+//
+// +k8s:deepcopy-gen:interfaces=ocm.software/open-component-model/bindings/go/runtime.Typed
+// +k8s:deepcopy-gen=true
+// +ocm:typegen=true
+type Config struct {
+	Type runtime.Type `json:"-"`
+	// IdleTimeout on startup. If the plugin is orphaned (e.g. due to a panic of the CLI)
+	// and a plugin is inactive for this duration, it will automatically shut itself down.
+	IdleTimeout Duration `json:"idleTimeout"`
+	// Locations is a list of locations where the plugin manager will look for plugins.
+	// This can be a list of directories.
+	Locations []string `json:"locations"`
+}
+
+type Duration time.Duration
+
+func (d Duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(time.Duration(d).String())
+}
+
+func (d *Duration) UnmarshalJSON(b []byte) error {
+	var v interface{}
+	if err := json.Unmarshal(b, &v); err != nil {
+		return err
+	}
+	switch value := v.(type) {
+	case float64:
+		*d = Duration(time.Duration(value))
+		return nil
+	case string:
+		tmp, err := time.ParseDuration(value)
+		if err != nil {
+			return err
+		}
+		*d = Duration(tmp)
+		return nil
+	default:
+		return errors.New("invalid duration")
+	}
+}
+
+// LookupConfig creates a new plugin configuration from a central V1 config.
+func LookupConfig(cfg *v1.Config) (*Config, error) {
+	var merged *Config
+	if cfg != nil {
+		cfg, err := v1.Filter(cfg, &v1.FilterOptions{
+			ConfigTypes: []runtime.Type{
+				runtime.NewVersionedType(ConfigType, Version),
+				runtime.NewUnversionedType(ConfigType),
+			},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to filter config: %w", err)
+		}
+		cfgs := make([]*Config, 0, len(cfg.Configurations))
+		for _, entry := range cfg.Configurations {
+			var config Config
+			if err := scheme.Convert(entry, &config); err != nil {
+				return nil, fmt.Errorf("failed to decode credential config: %w", err)
+			}
+			cfgs = append(cfgs, &config)
+		}
+		merged = Merge(cfgs...)
+		if merged == nil {
+			merged = &Config{}
+		}
+	} else {
+		merged = new(Config)
+	}
+
+	if len(merged.Locations) == 0 {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get user home directory to compute the default plugin directory: %w", err)
+		}
+		merged.Locations = []string{
+			filepath.Join(home, ".config", "ocm", "plugins"),
+		}
+	}
+
+	if merged.IdleTimeout == 0 {
+		merged.IdleTimeout = Duration(time.Hour)
+	}
+
+	return merged, nil
+}
+
+// Merge merges the provided configs into a single config.
+func Merge(configs ...*Config) *Config {
+	if len(configs) == 0 {
+		return nil
+	}
+
+	merged := new(Config)
+	_, _ = scheme.DefaultType(merged)
```

Comment:
Won't this overwrite different types to a single type or do I understand this incorrectly?

Reply:
this is for the merged configuration only. thats why I think using the default type for the merged config is fine.